cognite/client/data_classes/hosted_extractors/jobs.py

```diff
@@ -0,0 +1,478 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, ClassVar, Literal, cast
+
+from typing_extensions import Self, TypeAlias
+
+from cognite.client.data_classes._base import (
+    CogniteObject,
+    CognitePrimitiveUpdate,
+    CogniteResource,
+    CogniteResourceList,
+    CogniteUpdate,
+    ExternalIDTransformerMixin,
+    PropertySpec,
+    UnknownCogniteObject,
+    WriteableCogniteResource,
+    WriteableCogniteResourceList,
+)
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+TargetStatus: TypeAlias = Literal["running", "paused"]
+Status: TypeAlias = Literal[
```

Comment:
I suggest we name this `JobStatus`:
```suggestion
JobStatus: TypeAlias = Literal[
```